### ISSUE TITLE
disableservernat option crash restart kamailio

### ISSUE THREAD

**Issue by fouille** [dOpensource/dsiprouter#439]
Hello, 

When I wan't to disable natserver after enable, Kamailio can't restart. 
But if i re-enable natserver kamailio restart... 
Please see : 

```bash
root@siprouter-p1:/opt/dsiprouter# ./dsiprouter.sh disableservernat -debug                                                 
+ shift                                                                                                                    
+ ((  0 > 0  ))                                            
+ RUN_COMMANDS=($(printf '%s\n' "${RUN_COMMANDS[@]}" | awk '!x[$0]++'))
++ printf '%s\n' disableSERVERNAT
++ awk '!x[$0]++'
+ for RUN_COMMAND in "${RUN_COMMANDS[@]}"
+ disableSERVERNAT
+ disableKamailioConfigAttrib WITH_SERVERNAT /etc/dsiprouter/kamailio_dsiprouter.cfg
+ local NAME=WITH_SERVERNAT
+ local CONFIG_FILE=/etc/dsiprouter/kamailio_dsiprouter.cfg
+ sed -i -r -e 's~#+(!(define|trydef|redefine)[[:space:]]? WITH_SERVERNAT)~##\1~g' /etc/dsiprouter/kamailio_dsiprouter.cfg
+ printwarn 'SERVERNAT is disabled - Restarting Kamailio is required'
+ [[ SERVERNAT is disabled - Restarting Kamailio is required == \-\n ]]
+ printf '%b%s%b\n' '\033[1;33m' 'SERVERNAT is disabled - Restarting Kamailio is required' '\033[39;49;00m'
SERVERNAT is disabled - Restarting Kamailio is required
+ printdbg 'You can restart it by executing: systemctl restart kamailio'
+ [[ You can restart it by executing: systemctl restart kamailio == \-\n ]]
+ printf '%b%s%b\n' '\033[1;32m' 'You can restart it by executing: systemctl restart kamailio' '\033[39;49;00m'
You can restart it by executing: systemctl restart kamailio
+ cleanupAndExit 0
+ rm -f /etc/apt/apt.conf.d/local
+ set +x
```

KAMAILIO ERROR : 
```bash
root@siprouter-p1:/opt/dsiprouter# systemctl restart kamailio
Job for kamailio.service failed because the control process exited with error code.
See "systemctl status kamailio.service" and "journalctl -xe" for details.

root@siprouter-p1:/opt/dsiprouter# systemctl status kamailio.service
● kamailio.service - Kamailio (OpenSER) - the Open Source SIP Server
   Loaded: loaded (/lib/systemd/system/kamailio.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2022-06-20 21:19:36 CEST; 21s ago
  Process: 3358 ExecStart=/usr/sbin/kamailio -P /run/kamailio/kamailio.pid -f $CFGFILE -m $SHM_MEMORY -M $PKG_MEMORY (code=e

Jun 20 21:19:36 siprouter-p1 systemd[1]: kamailio.service: Service RestartSec=100ms expired, scheduling restart.
Jun 20 21:19:36 siprouter-p1 systemd[1]: kamailio.service: Scheduled restart job, restart counter is at 5.
Jun 20 21:19:36 siprouter-p1 systemd[1]: Stopped Kamailio (OpenSER) - the Open Source SIP Server.
Jun 20 21:19:36 siprouter-p1 systemd[1]: kamailio.service: Start request repeated too quickly.
Jun 20 21:19:36 siprouter-p1 systemd[1]: kamailio.service: Failed with result 'exit-code'.
Jun 20 21:19:36 siprouter-p1 systemd[1]: Failed to start Kamailio (OpenSER) - the Open Source SIP Server.
``` 

LOG KAMAILIO
```log
Jun 20 21:19:35 siprouter-p1 kamailio: INFO: <core> [core/tcp_main.c:5043]: init_tcp(): using epoll_lt as the io watch method (auto detected)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3347]: ERROR: tls [tls_init.c:845]: tls_check_sockets(): TLSs<MY_PUBLIC_IP:4443>: No listening socket found
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3347]: ERROR: <core> [core/sr_module.c:849]: init_mod(): Error while initializing module tls (/usr/lib/x86_64-linux-gnu/kamailio/modules/tls.so)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3347]: INFO: <core> [core/sctp_core.c:53]: sctp_core_destroy(): SCTP API not initialized
Jun 20 21:19:35 siprouter-p1 kamailio: INFO: <core> [core/tcp_main.c:5043]: init_tcp(): using epoll_lt as the io watch method (auto detected)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3351]: ERROR: tls [tls_init.c:845]: tls_check_sockets(): TLSs<MY_PUBLIC_IP:4443>: No listening socket found
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3351]: ERROR: <core> [core/sr_module.c:849]: init_mod(): Error while initializing module tls (/usr/lib/x86_64-linux-gnu/kamailio/modules/tls.so)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3351]: INFO: <core> [core/sctp_core.c:53]: sctp_core_destroy(): SCTP API not initialized
Jun 20 21:19:35 siprouter-p1 kamailio: INFO: <core> [core/tcp_main.c:5043]: init_tcp(): using epoll_lt as the io watch method (auto detected)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3354]: ERROR: tls [tls_init.c:845]: tls_check_sockets(): TLSs<MY_PUBLIC_IP:4443>: No listening socket found
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3354]: ERROR: <core> [core/sr_module.c:849]: init_mod(): Error while initializing module tls (/usr/lib/x86_64-linux-gnu/kamailio/modules/tls.so)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3354]: INFO: <core> [core/sctp_core.c:53]: sctp_core_destroy(): SCTP API not initialized
Jun 20 21:19:35 siprouter-p1 kamailio: INFO: <core> [core/tcp_main.c:5043]: init_tcp(): using epoll_lt as the io watch method (auto detected)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3357]: ERROR: tls [tls_init.c:845]: tls_check_sockets(): TLSs<MY_PUBLIC_IP:4443>: No listening socket found
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3357]: ERROR: <core> [core/sr_module.c:849]: init_mod(): Error while initializing module tls (/usr/lib/x86_64-linux-gnu/kamailio/modules/tls.so)
Jun 20 21:19:35 siprouter-p1 /usr/sbin/kamailio[3357]: INFO: <core> [core/sctp_core.c:53]: sctp_core_destroy(): SCTP API not initialized
Jun 20 21:19:36 siprouter-p1 kamailio: INFO: <core> [core/tcp_main.c:5043]: init_tcp(): using epoll_lt as the io watch method (auto detected)
Jun 20 21:19:36 siprouter-p1 /usr/sbin/kamailio[3360]: ERROR: tls [tls_init.c:845]: tls_check_sockets(): TLSs<MY_PUBLIC_IP:4443>: No listening socket found
Jun 20 21:19:36 siprouter-p1 /usr/sbin/kamailio[3360]: ERROR: <core> [core/sr_module.c:849]: init_mod(): Error while initializing module tls (/usr/lib/x86_64-linux-gnu/kamailio/modules/tls.so)
Jun 20 21:19:36 siprouter-p1 /usr/sbin/kamailio[3360]: INFO: <core> [core/sctp_core.c:53]: sctp_core_destroy(): SCTP API not initialized
``` 
`MY_PUBLIC_IP` is really my public ip ;-)

**Comment by tuxpowered** [comment]
I have not been able to get servernat to run period let alone try to disable it. 
What version of debian and dsip are you on?
What was your installation process? I can try to reproduce the issue for you and see if it is similar result.

**Comment by devopsec** [comment]
This looks like a known issue, the TLS memory issue in kamailio.  
Ref: https://github.com/dOpensource/dsiprouter/issues/459

This should be resolved in v0.721
